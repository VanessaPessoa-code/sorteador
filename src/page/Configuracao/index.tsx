import React from "react";
import Formulario from "../../components/Formulario";
import ListaParticipantes from "../../components/ListaParticipante";
import Rodape from "../../components/Rodape";
import Card from "../../components/Card";

const Configuracao = () => {

    return (
        <Card>
            <>
                <Formulario />
                <ListaParticipantes />
                <Rodape />
            </>
        </Card>
    )
}

export default Configuracao;