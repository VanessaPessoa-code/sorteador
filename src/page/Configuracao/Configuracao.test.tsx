import { render } from "@testing-library/react";
import exp from "node:constants";
import React from "react";
import { RecoilRoot } from "recoil";
import Configuracao from ".";

const mockNavegacao = jest.fn();
jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavegacao //hook que retorna uma função
    }
});

describe('a pagina de configuração', () => {
    test('deve ser renderizada corretamente' , () => {
        const {container} = render(
            <RecoilRoot>
                <Configuracao />
            </RecoilRoot>
        );

        expect(container).toMatchSnapshot();

    })
})