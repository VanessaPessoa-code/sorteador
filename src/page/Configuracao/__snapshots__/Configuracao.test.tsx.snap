// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuração deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="card"
  >
    <form
      class="form"
    >
      <h1>
         Vamos começar! 
      </h1>
      <div>
        <input
          class="campo_text"
          placeholder="Insira os nomes dos participantes"
          type="text"
          value=""
        />
        <button
          class="btn_add"
          disabled=""
          type="submit"
        >
           Adicionar 
        </button>
      </div>
      
    </form>
    <ul
      class="ul"
    />
    <footer
      class="rodape"
    >
      <button
        class="btn"
        disabled=""
      >
        <span>
          <img
            alt="Icone de play"
            src="/images/play_circle_outline.png"
          />
        </span>
        Iniciar brincadeira!
      </button>
      <img
        alt="Duas sacolas, verde e azul"
        src="/images/sacolas.png"
      />
    </footer>
  </div>
</div>
`;
